mango-core/src/main/java/org/calrissian/mango/domain/event/BaseEvent.java

```diff
@@ -16,66 +16,34 @@
 package org.calrissian.mango.domain.event;
 
 
+import com.google.common.base.Predicate;
+import com.google.common.collect.ArrayListMultimap;
 import com.google.common.collect.Multimap;
 import org.calrissian.mango.domain.Attribute;
 import org.calrissian.mango.domain.entity.BaseEntity;
 
-import static java.lang.System.currentTimeMillis;
-import static java.util.UUID.randomUUID;
+import java.util.Collection;
+import java.util.Set;
+
+import static com.google.common.collect.Collections2.filter;
 
 /**
  * Default implementation of {@link Event}
  */
 public class BaseEvent extends BaseEntity implements Event {
 
-    private final long timestamp; // in Millis
-
-    /**
-     * New event with random UUID and timestamp defaulted to current time
-     */
-    @Deprecated
-    public BaseEvent() {
-        this("", randomUUID().toString());
-    }   // for backward compatibility
-
-    /**
-     * New event with ID. Timestamp defaults to current time.
-     */
-    @Deprecated
-    public BaseEvent(String type, String id) {
-        this(type, id, currentTimeMillis());
-    }
-
-
-    @Deprecated
-    public BaseEvent(String id) {
-        this("", id, currentTimeMillis());
-    }
-
-    /**
-     * New store entry with ID and a timestamp
-     */
-    @Deprecated
-    public BaseEvent(String type, String id, long timestamp) {
-        super(type, id);
-        this.timestamp = timestamp;
-    }
-
-    @Deprecated
-    public BaseEvent(String id, long timestamp) {
-        this("", id, timestamp);
-    }
+    private static final String TIMESTAMP_FIELD = "_TIMESTAMP";
 
     BaseEvent(String type, String id, long timestamp, Multimap<String, Attribute> attributes) {
         super(type, id, attributes);
-        this.timestamp = timestamp;
+        addTimestamp(timestamp);
```

Comment:
Instead of adding the timestamp to the class attributes, add it to the arguement of the constructor before passing it up to the super constructor.  This way you can make the underlying class attributes map immutable.